cmd/di.go

```diff
@@ -726,35 +725,33 @@ func (di *Dependencies) bootstrapIdentityComponents(options node.Options) {
 }
 
 func (di *Dependencies) bootstrapDiscoveryComponents(options node.OptionsDiscovery) error {
+	di.DiscoveryStorage = discovery.NewStorage()
+
 	registry := discovery_composite.NewRegistry()
 	for _, discoveryType := range options.Types {
 		switch discoveryType {
 		case node.DiscoveryTypeAPI:
 			registry.AddRegistry(
 				discovery_api.NewRegistry(di.MysteriumAPI),
 			)
+			if !options.ProposalFetcherEnabled {
+				di.DiscoveryFinder = discovery_api.NewNoopFetcher()
+			} else {
+				di.DiscoveryFinder = discovery_api.NewFetcher(di.DiscoveryStorage, di.MysteriumAPI.Proposals, 30*time.Second)
+			}
 		case node.DiscoveryTypeBroker:
 			registry.AddRegistry(
 				discovery_broker.NewRegistry(di.BrokerConnection),
 			)
+			di.DiscoveryFinder = discovery_broker.NewProposalSubscriber(di.DiscoveryStorage, 61*time.Second, di.BrokerConnection)
```

Comment:
Why is this interval 61 second? Would be good to comment this behaviour as it is looks like that extra 1 second is needed for something.

Reply:
Clarified